### ISSUE TITLE
Bad Overlord indexing performance of files with events delimited by EOL character

### ISSUE THREAD

**Issue by l15k4** [druid-io/druid#2389]
Hey,

details are in this [discussion](https://groups.google.com/d/msg/druid-user/MVAArSZTriw/ZNFY3jVjBgAJ)

[Benchmark](http://i.imgur.com/1mJZngw.png)  <- the last FAILED benchmark is the same as the previous one except it is ingesting 50MB big files with lines (json events) delimited by EOLs (the way it should be done) ... the previous successful benchmarks are using 50MB big files each with just one HUGE line of all Json events.

Possibly Jackson performs way better with 50MB long line of Json than with 1600 lines of json?

**Comment by l15k4** [comment]
Btw I've been playing with the `implydata/distribution` for some time and I noticed of decreasing indexing performance with growing data sets.

You mention in the [benchmarking-druid](http://druid.io/blog/2014/03/17/benchmarking-druid.html) blog post, that : 

> The resulting Druid segments consist of:
>  - a single 589MB segment for the 1GB data set,
>  - 161 segments totaling 83.4GB (average segment size of 530MB) for the 100GB data set.
> 
> In our case the indexing service took about 25 minutes per segment for both datasets.

But from what I can see there is a linear decrease in ingestion speed as dataset grows. I tried to increase `Xmx` of relevant components but that doesn't help. The only component that actually does some CPU work and consumes memory is `middleManager`, the others are mostly idling... 

At first I thought that bottleneck is IO because I've been doing the benchmarks on Amazon EBS General Purpose (SSD) volumes, but then I tried physically attached SSD and it was 2x faster but the linear performance decrease was identical ...
